_drafts/gdb-python-api-pypi-packages.md

```diff
@@ -0,0 +1,453 @@
+ '/Users/tyler/venv/lib/python2.7/site-packages']
+```
+
+My shell's Python2 is from the Brew Python 2.7.15 installation and is importing
+packages from my virtual environment that I have activated. This is perfect.
+
+#### GDB
+
+Now we launch `arm-none-eabi-gdb-py`, start a Python shell, and check
+`sys.path`.
+
+```python
+(venv)
+$ arm-none-eabi-gdb-py
+GNU gdb (GNU Tools for Arm Embedded Processors 8-2018-q4-major) 8.2.50.20181213-git
+...
+(gdb) pi
+>>> import sys
+>>> sys.version
+'2.7.10 (default, Aug 17 2018, 19:45:58) \n[GCC 4.2.1 Compatible Apple LLVM 10.0.0 (clang-1000.0.42)]'
+>>> sys.path
+['/Users/tyler/.local/gcc-arm-none-eabi-8-2018-q4-major/arm-none-eabi/share/gdb/python',
+ '/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python27.zip',
+ '/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7',
+ '/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-darwin',
+ '/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-mac',
+ '/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-mac/lib-scriptpackages',
+ '/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-tk',
+ '/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-old',
+ '/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-dynload',
+ '/Library/Python/2.7/site-packages',
+ '/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python',
+ '/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/']
+```
+
+It looks like GDB is using the macOS System Python version 2.7.10 and is pulling
+in packages from the system installation at
+`'/Library/Python/2.7/site-packages'`. This is _wrong_, as it should be looking
+in our virtual environment's `site-packages/` directory.
+
+So how do we get the `sys.path` value within GDB's Python updated so I can
+import PrettyTable?!
+
+### Setting `sys.path` within `~/.gdbinit`
+
+We need three things:
+
+1. Find a way to modify GDB on initialization
+2. Extract our virtual environment's `sys.path` values
+3. append these paths to GDB's Python `sys.path` value.
+
+#### 1. Modifying GDB on launch
+
+To modify every instance of GDB that is launched, we can edit our `~/.gdbinit`
+file.
+
+Do note that GDB will look in the working directly for a `.gdbinit` file and
+that a file can be loaded using the `--init-command=FILE` argument when
+launching GDB. This is useful for working on team projects!
+
+#### 2. Extract `sys.path` values from `venv`
+
+To output the value of `sys.path` in a shell, we can actually launch Python in
+"command mode" and have it print a value to standard out. Below, we import `sys`
+and immediately print `sys.path` to standard out.
+
+```
+(venv)
+$ python -c "import sys;print(sys.path)"
+['/Users/tyler/.virtualenvs/memfault/lib/python3.6/site-packages', ...]
```

Comment:
is this from the wrong venv? I think I'd expect python2.x/site-packages?

Reply:
OOps, yeah, it's from the wrong env.